File: models/Position.js
const mongoose = require('mongoose');

const positionSchema = new mongoose.Schema({
  name: {
    type: String,
    required: [true, 'Position name is required'],
    unique: true,
    trim: true
  },
  description: {
    type: String,
    trim: true
  },
  isActive: {
    type: Boolean,
    default: true
  },
  isGlobal: {
    type: Boolean,
    default: false
  },
  governorate: {
    type: String,
    default: null,
    trim: true
  },
  createdBy: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User'
  }
}, {
  timestamps: true
});

// Index for performance (remove manual indexes to avoid duplicates)
positionSchema.index({ isActive: 1 });
positionSchema.index({ governorate: 1 });

module.exports = mongoose.model('Position', positionSchema);

File: mobile/lib/features/positions/domain/entities/position.dart
import 'package:equatable/equatable.dart';

class Position extends Equatable {
  final String id;
  final String name;
  final String? description;
  final bool isActive;
  final bool isGlobal;
  final String? governorate;
  final String? createdBy;
  final DateTime createdAt;
  final DateTime updatedAt;

  const Position({
    required this.id,
    required this.name,
    this.description,
    required this.isActive,
    required this.isGlobal,
    this.governorate,
    this.createdBy,
    required this.createdAt,
    required this.updatedAt,
  });

  @override
  List<Object?> get props => [
        id,
        name,
        description,
        isActive,
        isGlobal,
        governorate,
        createdBy,
        createdAt,
        updatedAt,
      ];
}